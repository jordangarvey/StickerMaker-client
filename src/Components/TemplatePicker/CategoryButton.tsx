import React, { FC } from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

import Categories from "./Categories";
import Shapes from "./Shapes";

import * as C from "../../Global/Colours";


/**
 * Interface for the Category Button props
 */
interface ICategoryButtonProps {
	/** The currently selected category this option belongs to e.g. shapes, text */
	category: Categories;
	/** The option that this button should display e.g. circle, square */
	option: Shapes;
	/** On press callback to be fired whenever the option button is pressed */
	onPress(option: Shapes): void;

	/** Optional boolean for whether or not this option is currently selected */
	selected?: boolean;
}

/**
 * Component to render a Category Button button
 */
const CategoryButton: FC<ICategoryButtonProps> = (props) => {
	return (
		<TouchableOpacity
			onPress={props.onPress.bind(null, props.option)}
			style={[
				styles.CategoryButton,
				props.selected && { backgroundColor: C.primary }
			]}
		>
			<Text style={props.selected && { color: C.textPrimary }}>
				{ props.option.charAt(0).toUpperCase() + props.option.slice(1) }
			</Text>
		</TouchableOpacity>
	);
};

/**
 * Styles for the Category Button button
 */
const styles = StyleSheet.create({
	CategoryButton: {
		alignItems: "flex-end",
		backgroundColor: C.secondary,
		borderRadius: 6,
		height: 80,
		flexDirection: "row",
		justifyContent: "center",
		marginHorizontal: 10,
		padding: 5,
		width: 60
	}
});

export default CategoryButton;
