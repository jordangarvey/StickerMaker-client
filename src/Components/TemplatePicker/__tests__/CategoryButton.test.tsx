import React from "react";

import renderer from "react-test-renderer";ies";

import CategoryButton from "../CategoryButton";


describe("The Category Button componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		const component = renderer.create(
			<CategoryButton
				category="shape"
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
