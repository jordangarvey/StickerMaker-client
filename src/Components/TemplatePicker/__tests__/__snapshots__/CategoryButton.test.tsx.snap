// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Category Button componentâ€™s snapshot renders as expected 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 6,
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "center",
      "marginHorizontal": 10,
      "opacity": 1,
      "padding": 5,
      "width": 60,
    }
  }
>
  <Text>
    Rect
  </Text>
</View>
`;
